proxy/agent.go

```diff
@@ -175,8 +175,14 @@ func (a *agent) Run(stop <-chan struct{}) {
 			}
 
 		case <-time.After(delay):
-			glog.V(2).Infof("Reconciling after delay %v", delay)
-			a.reconcile()
+			if reflect.DeepEqual(a.desired, a.current) {
+				glog.V(2).Info("Reconciliation not needed")
+				a.retryBudget = a.maxRetries
```

Comment:
reconcile already handles the DeepEqual, move reset of retryBudget and delay in reconcile, he and in the above case. They should all be in one place.